### PR TITLE
FEATURE: Autolinking to category using hashtags.

### PR DESCRIPTION
**Pull request by tgxworld**
- [x]  Need to disable autocomplete for `/^#?/`
- [x] Update preview when hash tag is added

### CODE REVIEW

**Comment by tgxworld** [comment]
I'll squash and merge once reviewed :smile:

File: app/assets/javascripts/discourse/components/composer-editor.js.es6 [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import userSearch from 'discourse/lib/user-search';
 import { default as computed, on } from 'ember-addons/ember-computed-decorators';
 import { linkSeenMentions, fetchUnseenMentions } from 'discourse/lib/link-mentions';
+import { linkSeenCategoryHashtags, fetchUnseenCategoryHashtags } from 'discourse/lib/link-category-hashtag';
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/components/composer-editor.js.es6]: This component should probably be name `link-category-hashtags`

File: app/assets/javascripts/discourse/components/d-editor.js.es6 [hunk 2]
```diff
@@ -175,7 +176,11 @@ export default Ember.Component.extend({
 
   @on('didInsertElement')
   _startUp() {
-    this._applyEmojiAutocomplete();
+    const container = this.get('container'),
+          $editorInput = this.$('.d-editor-input');
+
+    this._applyEmojiAutocomplete(container, $editorInput);
+    this._applyCategoyrHashtagAutocomplete(container, $editorInput);
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/components/d-editor.js.es6]: Typo `_applyCategoyr...`

**Comment by ZogStriP** [comment]
:+1:

**Comment by eviltrout** [comment]
This looks great! It's quite long though so we should keep a close on eye on it after merging.

**Comment by tgxworld** [comment]
Thanks for reviewing @ZogStriP and @eviltrout

**Pull request merged**